dlclive/dlclive.py

```diff
@@ -478,3 +486,154 @@ def close(self):
+                self.max_age, self.min_hits, self.sim_threshold,
+            )
+        else:
+            self.mot_tracker = trackingutils.SORTEllipse(
+                self.max_age, self.min_hits, self.sim_threshold,
+            )
+
+
+        data = {
+            "metadata": {
+                "all_joints_names": self.cfg["all_joints_names"],
+                "PAFgraph": self.cfg["partaffinityfield_graph"],
+                "PAFinds": self.cfg.get("paf_best", np.arange(self.n_multibodyparts))
+            }
+        }
+        # Hack to avoid IndexError when determining _has_identity
+        temp = {"identity": []} if self.identity_only else {}
+        data["frame0"] = temp
+        self.ass = inferenceutils.Assembler(
+            data,
+            max_n_individuals=self.n_animals,
+            n_multibodyparts=self.n_multibodyparts,
+            greedy=True,  # TODO Benchmark vs optimal matching
+            identity_only=self.identity_only,
+            max_overlap=1,
+        )
+
+        if frame is not None:
+            pose = self.get_pose(frame, **kwargs)
+        else:
+            pose = None
+
+        self.is_initialized = True
+
+        return pose
+
+    def get_pose(self, frame=None, **kwargs):
+        if frame is None:
+            raise DLCLiveError("No frame provided for live pose estimation")
+
+        frame = self.process_frame(frame)
+        data_dict = predict_multianimal.predict_batched_peaks_and_costs(
+            self.cfg,
+            np.expand_dims(frame, axis=0),
+            self.sess,
+            self.inputs,
+            self.outputs,
+        )[0]
+
+        assemblies, unique = self.ass._assemble(data_dict, ind_frame=0)
+        pose = np.full((self.n_animals, self.n_multibodyparts, 4), np.nan)
+        if self.n_animals == 1:
+            pose[0] = assemblies[0].data
+        else:
+            animals = np.stack([a.data for a in assemblies])
+            if not self.ass.identity_only:
+                if self.track_method == "box":
+                    xy = trackingutils.calc_bboxes_from_keypoints(animals)
+                else:
+                    xy = animals[..., :2]
+                trackers = self.mot_tracker.track(xy)[:, -2:].astype(np.int)
+            else:
+                # Optimal identity assignment based on soft voting
+                mat = np.zeros(
+                    (len(assemblies), self.n_animals)
+                )
+                for nrow, assembly in enumerate(assemblies):
+                    for k, v in assembly.soft_identity.items():
+                        mat[nrow, k] = v
+                inds = linear_sum_assignment(mat, maximize=True)
+                trackers = np.c_[inds][:, ::-1]
+            for i, j in trackers:
+                pose[i] = animals[j]
+        self.pose = (pose, unique)
+
+        if self.display is not None:
+            self.display.display_frame(frame, self.pose)
+
+        if self.resize is not None:
```

Comment:
missing corrections for `cropping` and `dynamic_cropping` which can't be set from `__init__`, but can be set as instance attributes. these should be split out into a separate method in the parent class to avoid having to duplicate code.